lib/internal/buffer.js

```diff
@@ -6,29 +6,33 @@ const {
   ERR_INVALID_ARG_TYPE,
   ERR_OUT_OF_RANGE
 } = require('internal/errors').codes;
-const { validateNumber } = require('internal/validators');
+const {
+  validateNumber,
+  isUint32
+} = require('internal/validators');
 const { setupBufferJS } = binding;
 
 // Remove from the binding so that function is only available as exported here.
 // (That is, for internal use only.)
 delete binding.setupBufferJS;
 
-// Temporary buffers to convert numbers.
-const float32Array = new Float32Array(1);
-const uInt8Float32Array = new Uint8Array(float32Array.buffer);
-const float64Array = new Float64Array(1);
-const uInt8Float64Array = new Uint8Array(float64Array.buffer);
+// Setup for internal dataview object
+const kIBDV = Symbol('internal buffer dataview');
+const dvWM = new WeakMap();
+const getIBDV = (ab) => dvWM.get(ab) ||
+  dvWM.set(ab, new DataView(ab)).get(ab);
 
-// Check endianness.
-float32Array[0] = -1; // 0xBF800000
-// Either it is [0, 0, 128, 191] or [191, 128, 0, 0]. It is not possible to
-// check this with `os.endianness()` because that is determined at compile time.
-const bigEndian = uInt8Float32Array[3] === 0;
+// Microoptimisations for benchmarking
+const _2_32 = 2 ** 32;
+const _2__32 = 2 ** -32;
+const _2_15 = 2 ** 15;
+const _2_7 = 2 ** 7;
```

Comment:
`2 ** n` should be constant fold. I doubt that it makes any difference to actually use these here.